functions/src/handle_friend_invitation.ts

```diff
@@ -82,56 +82,101 @@ export const handleFriendInvitation = functions.https.onRequest(
           (new Date().getTime() - inviteData.created.toDate().getTime()) /
           (1000 * 3600 * 24);
 
-        if (diffInDays >= 7) {
-          // Delete the invite record and the invite from users shcedule if link is expired
-          delete senderSchedule.terms[inviteData.term].versions[
-            inviteData.version
-          ].friends[inviteData.friend];
+        const defaultValidDuration = 7;
+        const validDuration = inviteData?.validFor
+          ? inviteData.validFor / (3600 * 24)
+          : defaultValidDuration;
+        if (diffInDays >= validDuration) {
+          // Check if invite link is for a specific friend
+          if (!inviteData.link && inviteData.friend) {
+            // Delete the invite record and the invite from users shcedule if single-invite link is expired
```

Comment:
Apologies for the confusion. An invite can either be a link OR be intended for a specific person. If an invite is a link, anyone can accept it.